### DISCUSSION TITLE
WAVE on PythonAnywhere

### DISCUSSION THREAD

**Discussion by fousasg** [h2oai/wave#1826] [General]
Hi, could WAVE be deployed in the PythonAnywhere platform? 
The PA says that any WSGI app can be deployed, and WAVE's documentation says that it is ASGI compatible. 
Any thoughts?

**Comment by mturoci** [comment]
Depends. While the Wave app server is a python ASGI-compatible server, it's not the server that is exposed to users via the URL. Our Go server (waved) is the one exposed and behaves like a proxy between the python app and the browser. If PythonAnywhere can expose a port, then Wave can be deployed without problems. If it can only run python apps, then it's not possible.
